packages/vats/src/orchestration.js

```diff
@@ -0,0 +1,268 @@
+// @ts-check
+/** @file Orchestration service */
+import { makeTracer } from '@agoric/internal';
+import { V as E } from '@agoric/vat-data/vow.js';
+import { M, matches } from '@endo/patterns';
+import { AmountShape, BrandShape } from '@agoric/ertp';
+import { makeICAConnectionAddress, parseAddress } from '@agoric/orchestration';
+import '@agoric/network/exported.js';
+
+const { Fail, bare } = assert;
+const trace = makeTracer('Orchestration');
+
+// XXX improve me
+/** @typedef {string} ChainAddress */
+
+/**
+ * @typedef {object} OrchestrationPowers
+ * @property {ERef<
+ *   import('@agoric/orchestration/src/types').AttenuatedNetwork
+ * >} network
+ */
+
+/**
+ * PowerStore is used so additional powers can be added on upgrade.
```

Comment:
please reference the ticket about Exo state migrations because the lack of that is what's motivating this.

the best issue I can find is https://github.com/Agoric/agoric-sdk/issues/7337